### PR TITLE
[KOGITO-8304] - Document the usage of all-in-one jobs-service image

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-8304

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -764,9 +764,46 @@ docker run -it --env SCRIPT_DEBUG=true quay.io/kiegroup/kogito-jobs-service-infi
...
 
 The ephemeral image does not have external dependencies like a backend persistence provider, it uses in-memory persistence
-while working with Jobs Services `allinone`, `infinispan`, `mongodb` and `postgresql` variants, it will need to have an Infinispan, MongoDB or PostgreSQL server,
-respectively, previously running.
+while working with Jobs Services `allinone`, `infinispan`, `mongodb` and `postgresql` variants, it will need to 
+have an Infinispan, MongoDB or PostgreSQL server, respectively, previously running.
 
+#### Jobs Services All-in-one 
+
+THe Jobs Services All in One image provides the option th run any supported variant that we have at disposal, which are:
```
Comment (spolti) [inline README.md]:
```suggestion
The Jobs Services All in One image provides the option th run any supported variant that we have at disposal, which are:
```

**Comment by kie-ci1** [comment]
**PR job** `#373` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/373/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
make: *** [clone-repos] Error 1<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f<br/>"docker rm" requires at least 1 argument.<br/>See 'docker rm --help'.<br/><br/>Usage:  docker rm [OPTIONS] CONTAINER [CONTAINER...]<br/><br/>Remove one or more containers<br/>+ date<br/>Thu Dec  1 09:58:05 EST 2022<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f<br/>"docker rmi" requires at least 1 argument.<br/>See 'docker rmi --help'.<br/><br/>Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]<br/><br/>Remove one or more images<br/>+ date<br/>Thu Dec  1 09:58:06 EST 2022<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/373/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: README.md [hunk 2]
```diff
@@ -764,9 +764,46 @@ docker run -it --env SCRIPT_DEBUG=true quay.io/kiegroup/kogito-jobs-service-infi
...
 
 The ephemeral image does not have external dependencies like a backend persistence provider, it uses in-memory persistence
-while working with Jobs Services `allinone`, `infinispan`, `mongodb` and `postgresql` variants, it will need to have an Infinispan, MongoDB or PostgreSQL server,
-respectively, previously running.
+while working with Jobs Services `allinone`, `infinispan`, `mongodb` and `postgresql` variants, it will need to 
+have an Infinispan, MongoDB or PostgreSQL server, respectively, previously running.
 
+#### Jobs Services All-in-one 
+
+The Jobs Services All in One image provides the option th run any supported variant that we have at disposal, which are:
```
Comment (radtriste) [inline README.md]:
```suggestion
The Jobs Services All in One image provides the option to run any supported variant that we have at disposal, which are:
```

File: README.md [hunk 3]
```diff
@@ -764,9 +764,46 @@ docker run -it --env SCRIPT_DEBUG=true quay.io/kiegroup/kogito-jobs-service-infi
...
+- MongoDB
+- Ephemeral (default if no variant is specified)
+
+There are 3 exposed environment variables that can be used to configure the behaviour, which are:
+
+- SCRIPT_DEBUG: enable debug level of the image and its operations
+- ENABLE_EVENTS: enable the events add-on
+- JOBS_SERVICE_PERSISTENCE: select which persistence variant to use
+
+Note, as the Jobs Services are built on top of Quarkus, we can also set any configured supported by Quarkus
```
Comment (radtriste) [inline README.md]:
```suggestion
Note: As the Jobs Services are built on top of Quarkus, we can also set any configuration supported by Quarkus
```

File: contrib/jobs-service/container-compose-infinispan.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    image: quay.io/kiegroup/kogito-jobs-service-allinone:latest
+    container_name: jobs-service
+    ports:
+      - "8080:8080"
+    depends_on:
+      infinispan:
+        condition: service_healthy
+    environment:
+      JOBS_SERVICE_PERSISTENCE: infinispan
+      KOGITO_JOBS_SERVICE_KNATIVE_EVENTS: "false"
```
Comment (radtriste) [inline contrib/jobs-service/container-compose-infinispan.yaml:26]: what is this env var ? do we need to set it by default ?

**Comment by kie-ci1** [comment]
**PR job** `#374` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/374/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
make: *** [clone-repos] Error 1<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f<br/>"docker rm" requires at least 1 argument.<br/>See 'docker rm --help'.<br/><br/>Usage:  docker rm [OPTIONS] CONTAINER [CONTAINER...]<br/><br/>Remove one or more containers<br/>+ date<br/>Thu Dec  1 12:49:12 EST 2022<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f<br/>"docker rmi" requires at least 1 argument.<br/>See 'docker rmi --help'.<br/><br/>Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]<br/><br/>Remove one or more images<br/>+ date<br/>Thu Dec  1 12:49:12 EST 2022<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/374/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci1** [comment]
**PR job** `#375` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/375/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
make: *** [clone-repos] Error 1<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f<br/>"docker rm" requires at least 1 argument.<br/>See 'docker rm --help'.<br/><br/>Usage:  docker rm [OPTIONS] CONTAINER [CONTAINER...]<br/><br/>Remove one or more containers<br/>+ date<br/>Thu Dec  1 12:50:02 EST 2022<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f<br/>"docker rmi" requires at least 1 argument.<br/>See 'docker rmi --help'.<br/><br/>Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]<br/><br/>Remove one or more images<br/>+ date<br/>Thu Dec  1 12:50:02 EST 2022<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/375/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: contrib/jobs-service/container-compose-infinispan.yaml (hunk 4 above)
Comment (spolti) [reply contrib/jobs-service/container-compose-infinispan.yaml:26]: I copied it from some example, it is not needed :)
Seems that it makes Jobs service to publish events. 
@tiagodolphine  ^

Comment (spolti) [reply contrib/jobs-service/container-compose-infinispan.yaml:26]: I am not sure if it is really needed**

Comment (tiagodolphine) [reply contrib/jobs-service/container-compose-infinispan.yaml:26]: this property is used to enable job-service to publish events to data-index through knative IIRC. So it is not related to any persistence, it is a common property for the job-service.
@wmedvede ˆ

File: contrib/jobs-service/sql/init.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
+CREATE ROLE "kogito-user" WITH
+    LOGIN
+    SUPERUSER
+    INHERIT
+    CREATEDB
+    CREATEROLE
+    NOREPLICATION
+    PASSWORD 'kogito-pass';
```
Comment (tiagodolphine) [inline contrib/jobs-service/sql/init.sql:8]: this would be used just for testing? I mean the credentials

Comment (spolti) [reply contrib/jobs-service/sql/init.sql:8]: it is meant to be as example for user to try.

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by cristianonicolai — APPROVED** [review]

**Comment by kie-ci1** [comment]
**PR job** `#376` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/376/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
make: *** [clone-repos] Error 1<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f<br/>"docker rm" requires at least 1 argument.<br/>See 'docker rm --help'.<br/><br/>Usage:  docker rm [OPTIONS] CONTAINER [CONTAINER...]<br/><br/>Remove one or more containers<br/>+ date<br/>Thu Dec  1 20:53:40 EST 2022<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f<br/>"docker rmi" requires at least 1 argument.<br/>See 'docker rmi --help'.<br/><br/>Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]<br/><br/>Remove one or more images<br/>+ date<br/>Thu Dec  1 20:53:40 EST 2022<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/376/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Review by radtriste — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -764,9 +764,46 @@ docker run -it --env SCRIPT_DEBUG=true quay.io/kiegroup/kogito-jobs-service-infi
...
+```bash
+podman run -it -e JAVA_OPTIONS='-Dmy.sys.prop1=value1 -Dmy.sys.prop2=value2' \
+  quay.io/kiegroup/kogito-jobs-service-allinone:latest
+```
+
+For convenience there are `container-compose` files that can be used to start the Jobs Service with the desired
+persistence variant, to use execute the following command:
+
+```bash
+podman-compose -f contrib/jobs-service/container-compose-<variant>.yaml
```
Comment (spolti) [inline README.md]:
```suggestion
podman-compose -f contrib/jobs-service/container-compose-<variant>.yaml up
```

**Comment by kie-ci1** [comment]
**PR job** `#377` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/377/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
make: *** [clone-repos] Error 1<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f<br/>"docker rm" requires at least 1 argument.<br/>See 'docker rm --help'.<br/><br/>Usage:  docker rm [OPTIONS] CONTAINER [CONTAINER...]<br/><br/>Remove one or more containers<br/>+ date<br/>Fri Dec  2 13:14:44 EST 2022<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f<br/>"docker rmi" requires at least 1 argument.<br/>See 'docker rmi --help'.<br/><br/>Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]<br/><br/>Remove one or more images<br/>+ date<br/>Fri Dec  2 13:14:45 EST 2022<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/377/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: contrib/jobs-service/container-compose-infinispan.yaml (hunk 4 above)
Comment (wmedvede) [reply contrib/jobs-service/container-compose-infinispan.yaml:26]: @tiagodolphine @spolti yep, this is correct. When we are in knative, and we want to send the JobStatus notifications events via knative eventing api to other services like the data-index, we set this configuration parameter to true.

File: README.md [hunk 7]
```diff
@@ -764,9 +764,46 @@ docker run -it --env SCRIPT_DEBUG=true quay.io/kiegroup/kogito-jobs-service-infi
...
+
+- PostgreSQL
+- Infinispan
+- MongoDB
+- Ephemeral (default if no variant is specified)
+
+There are 3 exposed environment variables that can be used to configure the behaviour, which are:
+
+- SCRIPT_DEBUG: enable debug level of the image and its operations
+- ENABLE_EVENTS: enable the events add-on
```
Comment (wmedvede) [inline README.md:782]: Is this variable ENABLE_EVENTS passed directly to the jobs service as a config property? or you are doing some translation to the jobs service "events-support" profile?

**Review by wmedvede — COMMENTED** [review]
I left a query, take a look please.
Thanks.

File: README.md (hunk 7 above)
Comment (spolti) [reply README.md:782]: @wmedvede please see here what this env does: https://github.com/kiegroup/kogito-images/blob/main/modules/kogito-jobs-service-common/added/launch/kogito-jobs-service-common.sh#L16

**Review by tiagodolphine — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (wmedvede) [reply README.md:782]: ok, thanks.

**Review by wmedvede — APPROVED** [review]

**Comment by spolti** [comment]
jenkins please retest this

**Pull request merged**